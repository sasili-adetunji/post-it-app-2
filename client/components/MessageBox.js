import React from 'react';
import { Card, CardHeader, CardTitle } from 'material-ui/Card';
import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import RaisedButton from 'material-ui/RaisedButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import PostItActions from '../actions/PostItActions';

const style = {
  height: 100,
  width: 100,
  margin: '30px 30px',
  padding: 30,
  textAlign: 'center',
  display: 'inline-block',
};

/**
 * Renders the MessageBox components
 *
 * @class MessageBox
 * @extends {React.Component}
 */
class MessageBox extends React.Component {

  /**
   * Creates an instance of MessageBox.
   * @param {any} props
   * @memberof MessageBox
   */
  constructor(props) {
    super(props);
    this.state = {
      message: '',
      groupId: '',
      priorityLevel: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }

  /**
   * monitors the state of the components state
   *
   * @param {any} e
   * @memberof MessageBox
   */
  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  /**
   * makes an action call to post message
   *
   * @param {any} e
   * @memberof MessageBox
   */
  onClick(e) {
    e.preventDefault();
    const message = {
      message: this.state.message,
      groupId: this.state.groupId,
      priorityLevel: this.state.priorityLevel,
    };
    PostItActions.addMessage(message);
  }

  /**
   * renders the messagebox components
   *
   * @returns {void}
   * @memberof MessageBox
   */
  render() {
    return (
         <MuiThemeProvider>
        <Card>
          <CardTitle title="Send Message" />
        <textarea
            name="message"
            value={this.state.message} onChange={this.onChange}
            style={{
              width: '200px',
              borderColor: '#D0D0D0',
              resize: 'none',
              borderRadius: 3,
              minHeight: '50px',
              color: '#555',
              fontSize: 14,
              outline: 'auto 0px'
            }} />
        <div>
          <TextField
          style={{
            width: '200px',
            borderColor: '#D0D0D0',
            resize: 'none',
            borderRadius: 3,
            minHeight: '50px',
            color: '#555',
            fontSize: 14,
            outline: 'auto 0px'
          }}
            name="groupId" onChange={this.onChange} value={this.state.groupId}
            floatingLabelText="Group ID" />
          <br />
          <label htmlFor="priorityLevel">Priority Level:</label>
          <select
            style={{
              width: '200px',
              borderColor: '#D0D0D0',
              resize: 'none',
              borderRadius: 3,
              minHeight: '50px',
              color: '#555',
              fontSize: 14,
              outline: 'auto 0px'
            }}
            placeholder="Priority Level" name="priorityLevel" onChange={this.onChange}
            value={this.state.priorityLevel} className="form-control">
            <option value="Normal">Normal</option>
            <option value="Urgent">Urgent</option>
            <option value="Critical">Critical</option>
          </select>
        </div>
        <RaisedButton
          style={{
            display: 'block',
            width: '20px'
          }}
            onClick={this.onClick}
            label="Send " primary
             />
        </Card>

            </MuiThemeProvider>
    );
  }

}

export default MessageBox;
