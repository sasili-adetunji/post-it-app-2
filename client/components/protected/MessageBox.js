import React from 'react';
import PostItActions from '../../actions/PostItActions';
import PostItStore from '../../stores/PostItStore';


/**
 * creates messagebox components
 *
 * @class MessageBox
 * @extends {React.Component}
 */
class MessageBox extends React.Component {

  /**
   * Creates an instance of MessageBox.
   * @param {any} props
   * @memberof MessageBox
   */
  constructor(props) {
    super(props);
    this.state = {
      message: '',
      priorityLevel: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }

  /**
   *
   * @param {any} e
   * @memberof MessageBox
   */
  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  /**
   *
   *
   * @param {any} e
   * @memberof MessageBox
   */
  onClick(e) {
    e.preventDefault();
    const message = {
      message: this.state.message,
      groupId: this.props.groupId.groupId,
      priorityLevel: this.state.priorityLevel,
      date: new Date().toJSON(),
      author: this.props.author.displayName
    };
    PostItActions.addMessage(message);
    this.setState({
      message: '',
      date: '',
    });
  }
  /**
   *
   * renders the messagebox components
   * @returns { void }
   * @memberof MessageBox
   */
  render() {
    return (
      <div className="input-group message_area">
        <div> <h4> Send Message to Group </h4> </div>
        <textarea
rows="2" onChange={this.onChange} id="btn-input"
        type="text" className="form-control"
          placeholder="Write your message here..." name="message" value={this.state.message} />
        <div />
        <select
            placeholder="Priority Level" name="priorityLevel" onChange={this.onChange}
            className="form-control" value={this.state.priorityLevel} >
          <option value="Normal">Normal</option>
          <option value="Urgent">Urgent</option>
          <option value="Critical">Critical</option>
        </select>
        <span className="input-group-btn">
          <div>
            <button onClick={this.onClick} className="btn btn-primary btn-sm">Send</button>
          </div>
        </span>
      </div>
    );
  }
}
export default MessageBox;
