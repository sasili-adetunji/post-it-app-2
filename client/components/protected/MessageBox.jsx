import React from 'react';
import PostItActions from '../../actions/PostItActions';
import PostItStore from '../../stores/PostItStore';
import * as Api from '../../Api';


/**
 * creates messagebox components
 *
 * @class MessageBox
 * 
 * @extends {React.Component}
 */
class MessageBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      message: '',
      priorityLevel: '',
      error: {}
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }


/**
* @method onChange
*
* @description Monitors changes in the components and change the state
*
* @memberof MessageBox
*
* @param {object} event
*
* @returns {void}
*/
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }


/**
 * @description Makes an action call to post message to a group
 * 
 * @param {object} event
 * 
 * @returns {void}
 * 
 * @memberof MessageBox
*/
  onClick(event) {
    event.preventDefault();
    if (!PostItStore.getOpenedGroup()[0]) {
      this.setState({
        error: { group: 'Please kindly select a group first' },
        message: ''
      });
    } else if (!this.state.message) {
      this.setState({
        error: { message : 'Kindly type your message first' }
      });
    } else if (!this.state.priorityLevel) {
       this.setState({
        error: { priorityLevel: 'priority level is required' }
      });
    }
     else {
      const message = {
        messageText: this.state.message,
        groupId: PostItStore.getOpenedGroup()[0].groupId,
        priorityLevel: this.state.priorityLevel,
        date: new Date().toJSON(),
        author: PostItStore.getLoggedInUser().data.userName
      };
      PostItActions.addMessage(message);
      this.setState({
        error: '',
        message: ''
      });
    }
  }
 /**
   * @method render
   * 
   * Render MessageBox component
   * 
   * @returns {String} The HTML markup for the MessageBox Components
   * 
   * @memberof MessageBox
   */
  
  render() {
    return (
      <div className='messageForm'>
        <div className='error'> 
          {this.state.error.message}  
          {this.state.error.priorityLevel} {this.state.error.group}   
        </div>
        <form onSubmit={this.onClick}>
          <div className='row'>
            <div className='col-md-10'>
              <div className="input-group">
                <input name='message'
                  placeholder='Enter a message' onChange={this.onChange} 
                  value={this.state.message} 
                  type="text" className="form-control messageInput"/> 
              </div>
            </div>
            <div className='col-md-2'>
              <div className="form-group">
                <select className="form-control selectPriority"
                  name="priorityLevel"onChange={this.onChange} 
                  value={this.state.priorityLevel} >
                    <option value="normal">Normal</option>
                    <option value="critical">Critical</option>
                    <option value="urgent">Urgent</option>
                </select>
              </div>
            </div> 
          </div>
        </form> 
      </div>
    );
  }
}
export default MessageBox;
