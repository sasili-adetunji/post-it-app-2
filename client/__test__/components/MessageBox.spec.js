import React from 'react';
import { shallow, render } from 'enzyme';
import MessageBox from '../../components/protected/MessageBox.jsx';


function setup() {
  const props = {
    onClick: () => {},
    onChange: () => {},
  };
  return shallow(<MessageBox {...props}/>);
}

describe('AddMember components', () => {
  const component = setup();

  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain two div', () => {
    expect(component.find('div').length).toEqual(2);
  });
  it('should recieve props', () => {
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
});

describe('MessageBox  Test', () => {
  it('should take props', () => {
    const component = setup();
    expect(component.props().onClick).toExist;
    expect(component.props().onChange).toExist;

  });
});
