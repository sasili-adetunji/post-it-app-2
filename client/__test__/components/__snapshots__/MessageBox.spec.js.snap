// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sendMessageDiv"
  >
    <strong
      className="error"
    >
       
      
       
    </strong>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group col-sm-2"
      >
        <select
          className="form-control"
          id="exampleFormControlSelect1"
          name="priorityLevel"
          onChange={[Function]}
          value=""
        >
          <option>
            Normal
          </option>
          <option>
            Urgent
          </option>
          <option>
            Critical
          </option>
        </select>
      </div>
      <input
        className="col-sm-10 sendMessageInput"
        name="message"
        onChange={[Function]}
        placeholder="Enter a message"
        value=""
      />
    </form>
  </div>,
  "nodes": Array [
    <div
      className="sendMessageDiv"
    >
      <strong
        className="error"
      >
         
        
         
      </strong>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group col-sm-2"
        >
          <select
            className="form-control"
            id="exampleFormControlSelect1"
            name="priorityLevel"
            onChange={[Function]}
            value=""
          >
            <option>
              Normal
            </option>
            <option>
              Urgent
            </option>
            <option>
              Critical
            </option>
          </select>
        </div>
        <input
          className="col-sm-10 sendMessageInput"
          name="message"
          onChange={[Function]}
          placeholder="Enter a message"
          value=""
        />
      </form>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <MessageBox
      onChange={[Function]}
      onClick={[Function]}
    />,
    "_instance": MessageBox {
      "context": Object {},
      "onChange": [Function],
      "onClick": [Function],
      "props": Object {
        "onChange": [Function],
        "onClick": [Function],
      },
      "refs": Object {},
      "state": Object {
        "error": "",
        "message": "",
        "priorityLevel": "",
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="sendMessageDiv"
    >
      <strong
        className="error"
      >
         
        
         
      </strong>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group col-sm-2"
        >
          <select
            className="form-control"
            id="exampleFormControlSelect1"
            name="priorityLevel"
            onChange={[Function]}
            value=""
          >
            <option>
              Normal
            </option>
            <option>
              Urgent
            </option>
            <option>
              Critical
            </option>
          </select>
        </div>
        <input
          className="col-sm-10 sendMessageInput"
          name="message"
          onChange={[Function]}
          placeholder="Enter a message"
          value=""
        />
      </form>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MessageBox
    onChange={[Function]}
    onClick={[Function]}
  />,
}
`;
